src/interface/report/hooks/useEvents.ts

```diff
@@ -15,19 +23,54 @@ const useEvents = ({
   player: PlayerInfo;
 }) => {
   const [events, setEvents] = useState<AnyEvent[] | null>(null);
+  // we manually paginate here. WCL could turn pagination back on in the WCL API but the amount loaded per page is too low
+  // for M+ to be usable without deeper changes on the WCL side.
+  const pageCount = fightPageCount(fight);
+  const [pagesLoaded, setPagesLoaded] = useState(0);
 
   useEffect(() => {
+    let cancelled = false;
+    const loadPage = async (startTime: number, endTime: number) => {
+      return await fetchEvents(report.code, startTime, endTime, player.id);
```

Comment:
Since this function is already `async`, don't need an `await` here. Technically don't need to make it `async` either, as it'll be a promise either way.